docs/standard/commandline/define-commands.md

```diff
@@ -0,0 +1,208 @@
+
+```console
+myapp --delay 21 --message "Hello world!"
+```
+
+```output
+--delay = 21
+--message = Hello world!
+```
+
+### Global options
+
+To add an option to one command at a time, use the `Add` or `AddOption` method as shown in the preceding example. To add an option to a command and recursively to all of its subcommands, use the `AddGlobalOption` method, as shown in the following example:
+
+:::code language="csharp" source="snippets/define-commands/csharp/Program2.cs" id="defineglobal" highlight="7" :::
+
+The preceding code adds `--delay` as a global option to the root command, and it's available in the handler for `subCommand1a`.
+
+## Define arguments
+
+[Arguments](syntax.md#arguments) are defined and added to commands like options. The following example is like the options example, but it defines arguments instead of options:
+
+:::code language="csharp" source="snippets/define-commands/csharp/Program2.cs" id="definearguments" :::
+
+Here's an example of command-line input and the resulting output for the preceding example code:
+
+```console
+myapp 42 "Hello world!"
+```
+
+```output
+<delay> argument = 42
+<message> argument = Hello world!
+```
+
+An argument that is defined without a default value, such as `messageArgument` in the preceding example, is treated as a required argument.  An error message is displayed, and the command handler isn't called, if a required argument isn't provided.
+
+## Define aliases
+
+Both commands and options support [aliases](syntax.md#aliases). You can add an alias to an option by calling `AddAlias`:
+
+```csharp
+var option = new Option("--framework");
+option.AddAlias("-f");
+```
+
+Given this alias, the following command lines are equivalent:
+
+```console
+myapp -f net6.0
+myapp --framework net6.0
+```
+
+Command aliases work the same way.
+
+```csharp
+var command = new Command("serialize");
+command.AddAlias("serialise");
+```
+
+This code makes the following command lines equivalent:
+
+```console
+myapp serialize
+myapp serialise
+```
+
+We recommend that you minimize the number of option aliases that you define, and avoid defining certain aliases in particular. For more information, see [Short-form aliases](syntax.md#short-form-aliases).
+
+## Required options
+
+To make an option required, set its `IsRequired` property to `true`, as shown in the following example:
+
+:::code language="csharp" source="snippets/define-commands/csharp/Program2.cs" id="requiredoption" :::
+
+The options section of the command help indicates the option is required:
+
+```output
+Options:
+  --endpoint <uri> (REQUIRED)
+  --version               Show version information
+  -?, -h, --help          Show help and usage information
+```
+
+If the command line for this example app doesn't include `--endpoint`, an error message is displayed and the command handler isn't called:
+
+```output
+Option '--endpoint' is required.
+```
+
+## Hidden commands, options, and arguments
```

Comment:
Somewhere lets add why you would do this:

The most common reason to do this is to continue to support a feature you would like to deprecate. It can also be used for admin or preview features.